### PR TITLE
LPD-25373 [Marketplace] Add Solutions Details Step Move Blocks

### PR DESCRIPTION
**Pull request by joanagrego**

### CODE REVIEW

**Comment by joanagrego** [comment]
ci:test:sf

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/context/SolutionContext.tsx [hunk 1]
```diff
@@ -18,6 +18,29 @@ import {ProductVocabulary} from '../enums/ProductVocabulary';
 import {useGetVocabulariesAndCategories} from '../hooks/data/useGetVocabulariesAndCategories';
 import HeadlessCommerceAdminCatalogImpl from '../services/rest/HeadlessCommerceAdminCatalog';
 
+export enum KebabDropdownItems {
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/context/SolutionContext.tsx:21]:
```suggestion
export enum BLOCK_DIRECTIONS {
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/context/SolutionContext.tsx [hunk 2]
```diff
@@ -18,6 +18,29 @@ import {ProductVocabulary} from '../enums/ProductVocabulary';
...
+	MOVE_TO_TOP = 'Move to Top',
+	MOVE_UP = 'Move Up',
+	MOVE_DOWN = 'Move Down',
+	MOVE_TO_BOTTOM = 'Move To Bottom',
+	DELETE = 'Delete',
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/context/SolutionContext.tsx:26]:
```suggestion
	MOVE_TO_TOP,
	MOVE_UP,
	MOVE_DOWN,
	MOVE_TO_BOTTOM,
	DELETE,
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/context/SolutionContext.tsx [hunk 3]
```diff
@@ -79,6 +92,7 @@ export enum SolutionTypes {
 }
 
 type SolutionPayload = {
+	[SolutionTypes.SET_BLOCK_MOVE]: {direction: string; index: number};
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/context/SolutionContext.tsx:95]:
```suggestion
	[SolutionTypes.SET_BLOCK_MOVE]: {direction: BLOCK_DIRECTIONS; index: number};
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/context/SolutionContext.tsx [hunk 4]
```diff
@@ -297,6 +311,46 @@ const reducer = (state: SolutionInitialState, action: AppActions) => {
...
+					blocks[newIndex] = blockToMove;
+				},
+				[KebabDropdownItems.DELETE]: () => {
+					blocks.splice(index, 1);
+				},
+			};
+
+			if (MoveOrDeleteBlockAction[direction]) {
+				MoveOrDeleteBlockAction[direction]();
+			}
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/context/SolutionContext.tsx:346]:
```suggestion
			MoveOrDeleteBlockAction[direction]();
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/context/SolutionContext.tsx [hunk 5]
```diff
@@ -297,6 +311,46 @@ const reducer = (state: SolutionInitialState, action: AppActions) => {
...
+					blocks.unshift(blockToMove);
+				},
+				[KebabDropdownItems.MOVE_TO_BOTTOM]: () => {
+					blocks.splice(index, 1);
+					blocks.push(blockToMove);
+				},
+				[KebabDropdownItems.MOVE_UP]: () => {
+					const newIndex = index - 1;
+					blocks[index] = blocks[newIndex];
+					blocks[newIndex] = blockToMove;
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/context/SolutionContext.tsx:332]:
```suggestion
					const newIndex = index - 1;
					
					blocks[index] = blocks[newIndex];
					blocks[newIndex] = blockToMove;
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/context/SolutionContext.tsx [hunk 6]
```diff
@@ -297,6 +311,46 @@ const reducer = (state: SolutionInitialState, action: AppActions) => {
...
+				},
+				[KebabDropdownItems.MOVE_DOWN]: () => {
+					const newIndex = index + 1;
+					blocks[index] = blocks[newIndex];
+					blocks[newIndex] = blockToMove;
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/context/SolutionContext.tsx:337]:
```suggestion
					const newIndex = index + 1;
					
					blocks[index] = blocks[newIndex];
					blocks[newIndex] = blockToMove;
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/context/SolutionContext.tsx [hunk 7]
```diff
@@ -297,6 +311,46 @@ const reducer = (state: SolutionInitialState, action: AppActions) => {
 			};
 		}
 
+		case SolutionTypes.SET_BLOCK_MOVE: {
+			const {direction, index} = action.payload;
+			const blocks = [...state.details];
+
+			const blockToMove = blocks[index];
+
+			const MoveOrDeleteBlockAction: MoveOrDeleteBlockActions = {
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/context/SolutionContext.tsx:320]:
```suggestion
			const blockActions: MoveOrDeleteBlockActions = {
```

**Review by kevenleone — CHANGES_REQUESTED** [review]

**Comment by kevenleone** [comment]
Just starting reviewing :)

**Comment by joanagrego** [comment]
Resent --> [PR](https://github.com/brianchandotcom/liferay-portal/pull/150214/files)

**Pull request closed without merging**